x-pack/legacy/plugins/reporting/server/lib/enqueue_job.ts

```diff
@@ -29,18 +25,16 @@ interface ConfirmedJob {
   _primary_term: number;
 }
 
-export function enqueueJobFactory(
+export async function enqueueJobFactory(
```

Comment:
For now, I'm forcing all these functions to be async to explicitly match the required type